nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-connection-configuration.js

```diff
@@ -1385,8 +1385,8 @@
 
                 // initialize the connection dialog
                 $.when(initializeSourceEditConnectionDialog(source), initializeDestinationEditConnectionDialog(destination, connection.destination)).done(function () {
-                    var availableRelationships = connection.availableRelationships;
-                    var selectedRelationships = connection.selectedRelationships;
+                    var availableRelationships = connection.availableRelationships.sort();
+                    var selectedRelationships = connection.selectedRelationships.sort();
```

Comment:
Both the available relationships and the selected relationships are sorted server side [1]. I think the behavior you're seeing is actually a bug in our cluster response merging logic. Since the data type of the DTO is just a `Set` when the object is unmarshalled the sorted ordered is lost. I think should be making sure the items are resorted after merging to ensure the responses from our REST API do not differ when standalone vs clustered.

[1] https://github.com/apache/nifi/blob/161e4b57632b2d5454390871980288eb66df2502/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/dto/DtoFactory.java#L702